doc/source/tune/user-guide.rst

```diff
@@ -295,49 +295,157 @@ When running a hyperparameter search, Tune can automatically and periodically sa
+      * Pytorch Lightning (:ref:`example <ray-lightning-tuning>`)
+      * Keras (:doc:`example </tune/examples/tune_mnist_keras>`)
+      * Tensorflow (:ref:`example <ray-train-tftrainer-example>`)
+      * LightGBM (:ref:`example <lightgbm-ray-tuning>`)
+  
+   2. **Custom training function**
+   
+      * All this means is that your function has to expose a ``checkpoint_dir`` argument in the function signature, and call ``tune.checkpoint_dir``. See :doc:`this example </tune/examples/custom_func_checkpointing>`, it's quite simple to do.
+
+Let's assume for this example you're running this script from your laptop, and connecting to your remote Ray cluster via ``ray.init()``, making your script on your laptop the "driver".
 
 .. code-block:: python
 
-    import os
-    import time
+    import ray
     from ray import tune
+    from your_module import my_trainable
+
+    ray.init(address="<cluster-IP>:<port>")  # set `address=None` to train on laptop
+
+    # configure how checkpoints are sync'd to the scheduler/sampler 
+    # we recommend cloud storage checkpointing as it survives the cluster when
+    # instances are terminated, and has better performance
+    sync_config = tune.syncConfig(
+        upload_dir="s3://my-checkpoints-bucket/path/",  # requires AWS credentials
+        sync_to_cloud=True,
+        sync_to_driver=False  # recommended for performance when using cloud checkpointing 
+    )
 
-    def train_func(config, checkpoint_dir=None):
-        start = 0
-        if checkpoint_dir:
-            with open(os.path.join(checkpoint_dir, "checkpoint")) as f:
-                state = json.loads(f.read())
-                start = state["step"] + 1
+    # here we wrap our Ray model (or custom training func) with a couple of things:
+    # 1) with_parameters() ensures when we checkpoint, weights are saved (not just hyperparameters)
+    # 2) durable() ensures that this trainer can be saved to cloud storage (ie: S3)
+    durable_trainer_with_params = tune.with_parameters(tune.durable(my_trainable))
 
-        for step in range(start, 100):
-            time.sleep(1)
+    # this starts the run!
+    tune.run(
+        durable_trainer_with_params,
+
+        # name of your experiment 
+        name="my-tune-exp",
+
+        # a directory where results are stored before being
+        # sync'd to head node/cloud storage
+        local_dir="/tmp/mypath",
 
-            # Obtain a checkpoint directory
-            with tune.checkpoint_dir(step=step) as checkpoint_dir:
-                path = os.path.join(checkpoint_dir, "checkpoint")
-                with open(path, "w") as f:
-                    f.write(json.dumps({"step": step}))
+        # see above! we will sync our checkpoints to S3 directory
+        sync_config=sync_config,
 
-            tune.report(hello="world", ray="tune")
+        # we'll checkpoint every 10 iterations, keep the best five
+        # checkpoints (by AUC score, descending), and always checkpoint
```

Comment:
```suggestion
        # checkpoints (by AUC score, reported by the trainable, descending), and always checkpoint
```

Reply:
what does `reported by the trainable` mean here?

Reply:
`max-auc` needs to be in `tune.report` inside the trainable (`tune.report(max-auc=score)`)